tests/cli-paths.rs

```diff
@@ -32,6 +32,10 @@ export PATH="$HOME/apple/bin"
         format!(". \"{dir}/{sh}\"\n", dir = dir, sh = sh)
     }
 
+    fn bad_source(dir: impl Display, sh: impl Display) -> String {
```

Comment:
`s/bad_source/source_pre_version_XYZ` ? Or `non_posix_source` ?
It's not easily clear why the function is called "bad".

Maybe extracting `&bad_source("$HOME/.cargo", POSIX_SH)` as a constant and adding a comment to make it clear what it means would also help down the line?

Reply:
Relabelled the function `non_posix_source` which I think is probably good enough.

I also added an expositionary comment as to why this is present.